utilities/src/main/java/io/syndesis/qe/common/CommonSteps.java

```diff
@@ -77,49 +83,48 @@ public static void undeploySyndesis() {
+     * Since Syndesis endpoints uses old kubernetes-model, the CRD is missing (`NoClassDefFoundError`)
+     * When we let Syndesis endpoints to bring the old kubernetes-model with itself (remove <exclude> from parent pom.xml),
+     * it causes undesired behaviour, e.g. portForwarding doesn't work correctly etc.
+     * So we need to wait with bump xtf version until the Syndesis will contains newer kubernetes-model
+     * <p>
+     * This is implementation from XTF with workaround
+     * To access to the protected methods, the JavaReflection API is used.
+     */
+    private static Waiter isProjectClean() {
+        return new SimpleWaiter(() -> {
+            int crdInstances = 0;
+            int removableResources = 0;
+            try {
+                Method privateMethodGetCRDContextProviders = OpenShift.class.getDeclaredMethod("getCRDContextProviders", null);
+                privateMethodGetCRDContextProviders.setAccessible(true);
+                Method privateMethodListRemovableResources = OpenShift.class.getDeclaredMethod("listRemovableResources", null);
+                privateMethodListRemovableResources.setAccessible(true);
+                ServiceLoader<CustomResourceDefinitionContextProvider> cRDContextProviders =
+                    (ServiceLoader<CustomResourceDefinitionContextProvider>) privateMethodGetCRDContextProviders.invoke(OpenShift.class,
+                        null);
+                for (CustomResourceDefinitionContextProvider crdContextProvider : cRDContextProviders) {
+                    try {
+                        crdInstances += ((List) (OpenShiftUtils.getInstance().customResource(crdContextProvider.getContext())
+                            .list(OpenShiftUtils.getInstance().getNamespace()).get("items"))).size();
+                    } catch (KubernetesClientException kce) {
+                        // CRD might not be installed on the cluster
+                    }
+                }
+                List<HasMetadata> removableResourcesList =
+                    (List<HasMetadata>) privateMethodListRemovableResources.invoke(OpenShiftUtils.getInstance(), null);
+                removableResources = removableResourcesList.size();
+            } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+                e.printStackTrace();
+            }
+            log.info(String.format("Testing if project is clean, CRD instances: %d , RemovableResources: %d", crdInstances, removableResources));
+            return crdInstances == 0 &
+                removableResources <= 3;
+            // +1 because configMap can be there on OCP 4.7, see https://github.com/xtf-cz/xtf/issues/406 ,
+            // +2 resources from jaeger operator if there was a jaeger in this namespace before and there is another namespace with jaeger operator
+        }, TimeUnit.MILLISECONDS, WaitingConfig.timeoutCleanup(), "Cleaning project.");
```

Comment:
@mkralik3 I think this is no longer needed right? I might've included it in the backport by accident

Reply:
@mmuzikar  correct, after XTF update, this isn't longer needed.